src/compiler/main.ml

```diff
@@ -488,6 +488,9 @@ try
 	Common.define_value com Define.Haxe (s_version false);
 	Common.define_value com Define.Dce "std";
 	com.warning <- (fun msg p -> message ctx (CMWarning(msg,p)));
+	com.get_warnings <- (fun () -> (List.map compiler_message_string (List.rev ctx.messages)));
+	com.clear_warnings <- (fun () -> (ctx.messages <- []));
```

Comment:
This line will prune errors and info too. `ctx.messages` stores all kind of compiler messages, not only warnings.

Reply:
Perhaps the API should actually allow to access/edit all messages.

Reply:
Hmm I'll do some more testing tonight/tomorrow.

I'm not sure it can access/edit all messages (fatal errors or something might prevent it from doing so?), but I can at least filter messages so that this API only manipulates warnings like it says it does.

Reply:
@RealyUniqueName I updated to only manipulate warnings, but I'm not sure how to test it (if I have an error, `Context.xWarnings()` won't execute)